src/submission/submission-service.ts

```diff
@@ -891,23 +895,55 @@ export namespace operations {
           .processParallel(command.clinicalType, record, index, schema);
 
         if (schemaResult.validationErrors.length > 0) {
+          // check for program exception
           const result = await programExceptionRepository.find(command.programId);
 
-          // filter out valid exceptions before adding to error accumulator
-          const validationErrors =
-            // only filter is we have valid exceptions available
-            isProgramException(result)
-              ? schemaResult.validationErrors.filter(validationError => {
-                  return !applyExceptionIfExists(result.exceptions, validationError, record);
-                })
-              : schemaResult.validationErrors;
+          let validationErrors: DeepReadonly<SchemaValidationError>[] = [];
+
+          if (isProgramException(result)) {
+            schemaResult.validationErrors.forEach(validationError => {
+              const validationErrorFieldName = validationError.fieldName;
+              const recordValue = record[validationErrorFieldName];
+
+              // Zero Array type exceptions exist, but recordValue type is string | string[]
+              if (Array.isArray(recordValue)) {
+                validationErrors.push(validationError);
+                return;
+              }
```

Comment:
there are no record field values or exception values that are currently arrays but this accounts for the typing coming from lectern external lib and future consideration.